src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -795,31 +795,21 @@ public interface IObjectCreationExpression : IExpression
         /// <summary>
         /// Explicitly-specified member initializers.
         /// </summary>
-        ImmutableArray<IMemberInitializer> MemberInitializers { get; }
+        ImmutableArray<ISymbolInitializer> MemberInitializers { get; }
     }
 
     /// <summary>
     /// Represents an object member initializer.
     /// </summary>
-    public interface IMemberInitializer : IOperation
+    public interface ISymbolInitializer : IOperation
     {
-        MemberInitializerKind MemberInitializerKind { get; }
         IExpression Value { get; }
     }
-
-    /// <summary>
-    /// Kinds of member initializers.
-    /// </summary>
-    public enum MemberInitializerKind
-    {
-        Field,
-        Property
-    }
-
+    
     /// <summary>
     /// Represents an initialization of a field.
     /// </summary>
-    public interface IFieldInitializer : IMemberInitializer
+    public interface IFieldInitializer : ISymbolInitializer
```

Comment:
I have wrestled with this question for a few days. The choices, as I understand them:
  1) Leave the API as is, with only a property that returns the first symbol. This is completely correct for C#, and works correctly for the vast majority of VB programs. However, for VB programs that initialize multiple fields with As New or array types with sizes, the behavior is incorrect with no workaround (except for falling back on operation block actions).
  2) Have both a property that returns the first symbol and one that returns all symbols. In practice, this is as bad as 1) because almost nobody will write analyzers that examine all the symbols and the additional property will be confusing.
  3) Have only a property that returns all the symbols. This will confuse and annoy almost all analyzer authors, but does make it more likely that analyzers will be correct.

3) is not satisfying but is the only defensible choice.

Anyone care to argue a different position?